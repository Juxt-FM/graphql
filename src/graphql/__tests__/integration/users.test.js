const { createTestClient } = require("apollo-server-testing");
const { gql } = require("apollo-server-express");
const { buildTestServer } = require("../__utils");

const { mockUser } = require("../../sources/__tests__/users.test");

const GET_CURRENT_USER = gql`
  query {
    me {
      id
      email {
        address
        verified
      }
      phone {
        number
        verified
      }
      profile {
        name
      }
      lastLogin
      updatedAt
      createdAt
    }
  }
`;

describe("User Queries", () => {
  it("fetch authenticated user account", async () => {
    const { server, mockUserService } = await buildTestServer();

    mockUserService.getById.mockReturnValueOnce({
      ...mockUser,
      id: mockUser._id,
    });

    const { query } = createTestClient(server);
    const res = await query({ query: GET_CURRENT_USER });

    expect(res).toMatchSnapshot();
  });
});

const UPDATE_USER = gql`
  mutation UpdateUser($data: UpdateUserInput!) {
    updateUser(data: $data) {
      id
      profile {
        name
      }
      email {
        address
        verified
      }
      phone {
        number
        verified
      }
      lastLogin
      updatedAt
      createdAt
    }
  }
`;

describe("User Mutations", () => {
  it("update user account", async () => {
    const { server, mockUserService } = await buildTestServer();

    mockUserService.update.mockReturnValueOnce({
      user: { ...mockUser, id: mockUser._id },
    });

    const { mutate } = createTestClient(server);
    const res = await mutate({
      mutation: UPDATE_USER,
      variables: {
        data: {
          name: "Test User",
          email: "new@email.com",
          phoneNumber: "+11234567890",
        },
      },
    });

    expect(res).toMatchSnapshot();
  });
});
