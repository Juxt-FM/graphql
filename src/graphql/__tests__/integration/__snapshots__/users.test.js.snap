// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Mutations update user account 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "createdAt": "1604465235451",
      "email": Object {
        "address": "user@email.com",
        "verified": true,
      },
      "id": "1",
      "lastLogin": "1604465235451",
      "phone": null,
      "profile": Object {
        "name": "Test User",
      },
      "updatedAt": "1604465235451",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`User Queries fetch authenticated user account 1`] = `
Object {
  "data": Object {
    "me": Object {
      "createdAt": "1604465235451",
      "email": Object {
        "address": "user@email.com",
        "verified": true,
      },
      "id": "1",
      "lastLogin": "1604465235451",
      "phone": null,
      "profile": Object {
        "name": "Test User",
      },
      "updatedAt": "1604465235451",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
